### PR TITLE
Proposals: you asked, so here you go

### PR DESCRIPTION
**Pull request by hagemt**
Off to a good start. I just added a simple README and set up my usual boilerplate.

You can add `jest` or whatever and attach that to `npm test`, if you'd rather use that.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  extends: [
+    'eslint:recommended',
+    'plugin:import/recommended',
+    'plugin:mocha/recommended',
+    'plugin:node/recommended',
+    'plugin:promise/recommended',
+    'prettier-standard'
+  ],
+  parserOptions: {
+    ecmaVersion: 2017
```
Comment (hagemt) [inline .eslintrc.js:15]: This is necessary for `eslint` to understand async/await.

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  plugins: [
+    'import',
+    'mocha',
+    'node',
+    'prettier',
+    'promise', // needed for standard
+    'standard'
+  ],
+  rules: {
```
Comment (hagemt) [inline .eslintrc.js:25]: I'd really like to add `'comma-dangle': ['warn', 'always-multiline'],`, but `standard` doesn't seem to agree.

File: package.json [hunk 3]
```diff
@@ -1,19 +1,47 @@
...
+  "keywords": [],
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cpnota/rl-js-agents.git"
+  },
+  "scripts": {
+    "coverage": "nyc mocha # writes to .nyc_output and prints report to console",
+    "eslint": "eslint .*.js *.js **/*.js --report-unused-disable-directives",
+    "postreport": "npm run eslint -- --max-warnings 0 # what's the policy?",
+    "prereport": "npm run eslint -- --fix # runs prettier-standard",
```
Comment (hagemt) [inline package.json:43]: This is my latest recommendation: could be moved to pretest/posttest. Always test/lint pretty code.

File: package.json [hunk 4]
```diff
@@ -1,19 +1,47 @@
 {
   "name": "rl-js-agents",
-  "version": "0.0.1",
+  "version": "0.1.0",
   "description": "Reinforcement learning agents in JavaScript",
-  "author": "Chris Nota <cnota@cs.umass.edu>",
```
Comment (hagemt) [inline package.json:5]: FYI, `npm` parses the `AUTHORS` file, which will scale-up much better than `package.json`.

File: .eslintrc.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+module.exports = {
+  env: {
+    es6: true,
+    node: true
```
Comment (hagemt) [inline .eslintrc.js:4]: I usually don't add this here (prefer to do it per module) but this project is 100% `node`.

File: .eslintrc.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  },
+  extends: [
+    'eslint:recommended',
+    'plugin:import/recommended',
+    'plugin:mocha/recommended',
+    'plugin:node/recommended',
+    'plugin:promise/recommended',
+    'prettier-standard'
```
Comment (hagemt) [inline .eslintrc.js:12]: I'm glad this exists, because otherwise getting these to play nicely seems like a PITA.

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  parserOptions: {
+    ecmaVersion: 2017
+  },
+  plugins: [
+    'import',
+    'mocha',
+    'node',
```
Comment (hagemt) [inline .eslintrc.js:20]: I've never used this plugin before, but I can see why one would.

File: README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+## Getting Started
+
+This is a [`node`](https://nodejs.org) module that will be made available via GitHub and [`npm`](https://www.npmjs.com/).
+
+Check dependencies and `engines`, etc. in `package.json`; this module targets the current LTS, and possibly higher versions.
+
+Since nothing has been published to `npm` so far, to start hacking: clone this repo with `git` and therein run `npm install`.
+
+Use `npm test` to run the standard (fast) test suite; `npm run report` will generate a comprehensive test artifact.
```
Comment (hagemt) [inline README.md:13]: Right now, `npm test` just points to `npm run report`. Add `"test": "npm run jest",` or whatever, to taste.

File: README.md [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+
+Versioning follows [`semver`](https://semver.org/spec/v2.0.0.html) with odd-numbered minor versions for development releases.
+
+The project welcomes PRs with continuous integration and [greenkeeper.io](https://greenkeeper.io/), among other great tools.
```
Comment (hagemt) [inline README.md:17]: I very much recommend this tool, especially if you fix versions of your dependencies in `package.json`.

File: package.json [hunk 10]
```diff
@@ -1,19 +1,47 @@
...
-    "eslint-plugin-import": "^2.8.0",
-    "eslint-plugin-node": "^5.2.1",
-    "eslint-plugin-prettier": "^2.4.0",
-    "eslint-plugin-promise": "^3.6.0",
-    "eslint-plugin-standard": "^3.0.1",
-    "prettier": "^1.10.1"
+    "eslint": "4.15.0",
+    "eslint-config-prettier": "2.9.0",
+    "eslint-config-prettier-standard": "1.0.1",
+    "eslint-config-standard": "10.2.1",
```
Comment (hagemt) [inline package.json:17]: I got a warning about a peer dependency here. Seems like v11 isn't out of beta.

**Review by hagemt — COMMENTED** [review]
I frankly don't understand using `yarn` these days.

Native `npm install` is 2-4x faster w/ modern `npm`.

**Comment by cpnota** [comment]
Okay fine, dropping yarn

**Pull request closed without merging**